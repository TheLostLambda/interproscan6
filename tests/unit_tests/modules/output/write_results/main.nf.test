nextflow_process {

    name "Test Process WRITE_RESULTS"
    script "interproscan/modules/output/write_results/main.nf"
    process "WRITE_RESULTS"

    stage {
        copy "interproscan/scripts/output/write_results.py"
        copy "interproscan/scripts/output/format_writer/regex.py"
        copy "interproscan/scripts/output/format_writer/tsv_output.py"
        copy "interproscan/scripts/output/format_writer/xml_output.py"
        copy "interproscan/scripts/output/format_writer/json_output.py"
        copy "tests/unit_tests/test_inputs/write_results/protein/matches.json"
        copy "tests/unit_tests/test_inputs/write_results/protein/parsed_seqs.json"
    }

    test("Should pass") {

        when {
            params {
            }
            process {
                """
                input[0] = "filename"
                input[1] = "../../../meta/tests/unit_tests/test_inputs/write_results/protein/parsed_seqs.json"
                input[2] = file("meta/tests/unit_tests/test_inputs/write_results/protein/matches.json")
                input[3] = "tsv"
                input[4] = "outpath"
                input[5] = "unittest.version"
                input[6] = "false"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
