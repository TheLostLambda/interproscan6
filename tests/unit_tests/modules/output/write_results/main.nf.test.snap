{
    "Should pass": {
        "content": [
            {
                "0": [
                    "filename.ips6.tsv:md5,a8bda595d33252c0b82763c08fa260c5"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-09T09:11:19.71282"
    }
}